src/fondue/Fondue.js

```diff
@@ -22,6 +23,39 @@ export default class Fondue {
 		return this.colorFormats.length >= 1;
 	}
 
+	// Return an object of all language systems supported by
+	// either GSUB or GPOS. Tags are stripped ("ROM " → "ROM").
+	get languageSystems() {
+		const getLangs = (table) => {
+			let langs = [];
+			if (table) {
+				const scripts = table.getSupportedScripts();
+				for (const script of scripts) {
+					const scriptTable = table.getScriptTable(script);
+					const languages = table.getSupportedLangSys(scriptTable);
+					if (languages.length) {
+						for (const language of languages) {
+							langs.push(language.trim());
+						}
+					}
+				}
+			}
+			return langs;
+		};
+
+		const gsubLangs = getLangs(this._font.opentype.tables.GSUB);
+		const gposLangs = getLangs(this._font.opentype.tables.GPOS);
+		const allLangs = new Set([...gsubLangs, ...gposLangs]);
+
+		let langSys = {};
+		for (const lang of allLangs) {
+			if (languages[lang]) {
+				langSys[lang] = languages[lang];
+			}
+		}
+		return langSys;
```

Comment:
This could also be written in a more functional style as follows:

```js
return [...allLangs]
	.filter((lang) => languages[lang])
	.reduce((acc, lang) => ({ ...acc, [lang]: languages[lang] }), {});
```

In most cases a more functional style is more readable IMO, because it focuses on the data pipeline, but in this case I'm not so sure 😅 . WDYT?

Reply:
I was dissatisfied with my implementation, it seems so verbose and overcomplicated for such small functionality. But at least I see what's going on, to be honest my internal JavaScript parser has trouble understanding your suggestion 😅  Let's leave the dumb code in for now?